api/src/types.rs

```diff
@@ -275,6 +290,147 @@ pub struct Backup {
     pub region: String,
     pub status: String,
     pub created: String,
+    pub linked_resource: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JobDataScheduledBackup {
+    pub description: String,
+    pub max_backup_count: i32,
+    pub cluster_id: ClusterId,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OtherJobData {
+    pub r#type: String,
+    pub data: serde_json::value::Value,
+}
+
+#[derive(Debug, Serialize)]
+#[serde(tag = "type", content = "data")]
+pub enum JobData {
+    ScheduledBackup(JobDataScheduledBackup),
+    Other(OtherJobData),
+}
+
+impl<'de> serde::Deserialize<'de> for JobData {
```

Comment:
With that change I proposed above, you don't need to do that anymore and the code would still look high level enough.